packages/flutter/lib/src/material/date.dart

```diff
@@ -0,0 +1,225 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show hashValues;
+
+import 'package:flutter/foundation.dart';
+
+import 'material_localizations.dart';
+
+/// Utility functions for working with dates.
+class DateUtils {
+  // This class is not meant to be instantiated or extended; this constructor
+  // prevents instantiation and extension.
+  DateUtils._();
+
+  /// Returns a [DateTime] with just the date of the original, but no time set.
```

Comment:
It's not completely clear what "no time set" means.  I assume that it means that all of the time will be zero.

"no time set" is used quite a bit in the API doc that follows, so maybe just define the expression in the class API doc.